src/main/java/gift/product/domain/ProductOption.java

```diff
@@ -0,0 +1,70 @@
+package gift.product.domain;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+
+@Entity
+public class ProductOption {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, length = 50)
+    private String name;
+
+    @Column(nullable = false)
+    private Integer quantity;
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "product_id", nullable = false)
+    private Product product;
+
+    protected ProductOption() {
+
+    }
+
+    public ProductOption(String name, Integer quantity){
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public void setProduct(Product product) {
+        if(this.product != null) {
+            this.product.getProductOptions().remove(this);
+        }
+        this.product = product;
+
+        if(product != null && !product.getProductOptions().contains(this)) {
+            product.getProductOptions().add(this);
+        }
```

Comment:
```suggestion
        if(this.product != null) {
            this.product.remove(this);
        }
        this.product = product;

        if(product != null && !product.getProductOptions().contains(this)) {
            product.addOption(this);
        }
```

이런 형태로 고민하는 방법에 대해 고민해보면 좋을 것 같아요-! 
getter로 꺼내서 사용하게 되면 응집도가 낮아지고 어떻게 사용하는 것인지에 대해 추적하기가 어려워져요-!

Reply:
엔티티 객체 내에서 처리가능한 로직은 객체 내에서 처리하도록 구현해야 하는데, 아직 getter, setter를 맹목적으로 사용하던 습관이 좀 남아 있는 것 같습니다. 이 부분도 함께 수정해보겠습니다.